rfc/previews-using-fetaure-flags.md

```diff
@@ -0,0 +1,306 @@
+    steps:
+      # 1. Create & Publish feature subgraph based on cosmo.yaml
+      # 2. Create a new feature flag
+      - uses: wundergraph/feature-flags@main
+        id: ff
+        with:
+          config: cosmo.yaml # Default to .github/cosmo.yaml
+          create: true # Create the feature flag + subgraphs from cosmo.yaml
+      - run: |
+          echo "Feature flag has been created. You can now provision the subgraph service."
+          echo "Published Feature Subgraphs: ${{ steps.ff.outputs.published_feature_subgraphs }}"
+
+  # run the job to deploy all the changed subgraphs here.
+
+  update:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'synchronize'
+    steps:
+      - uses: wundergraph/feature-flags@main
+        with:
+          config: cosmo.yaml # Default to .github/cosmo.yaml
+          update: true # Only update the feature subgraphs from cosmo.yaml
+      - run: |
+          echo "Feature subgraph has been updated. You can now update the subgraph service."
+          echo "Published Feature Subgraphs: ${{ steps.ff.outputs.published_feature_subgraphs }}"
+
+  destroy:
+    if: github.event.pull_request.action == 'closed' || (github.event.label.name == 'cosmo-preview' && github.event.pull_request.action == 'unlabeled')
+    runs-on: ubuntu-latest
+    steps:
+      - uses: wundergraph/feature-flags@main
+        id: ff
+        with:
+          config: cosmo.yaml # Default to .github/cosmo.yaml
+          destroy: true # Only destroy the feature flag + subgraphs from cosmo.yaml
+      - run: |
+          echo "Feature flag and the feature subgraphs have been destroyed. You can now unprovision the subgraph service."
+---
+# config
+# .github/cosmo.yaml
+namespace: 'staging'
+feature_flags:
+  - name: 'my-feature-flag-{PR_NUMBER}'
+    labels:
+      - 'my-feature-flag-label'
+
+subgraphs:
+  - name: 'subgraph1'
+    schema: 'my-project/my-subgraph/subgraph1.graphql'
+    routing_url: 'http://mycompany-subgraph1-{PR_NUMBER}.com/graphql' # Supports templating, only PR_NUMBER is supported
+```
+
+### Requirements
+
+- Make sure the labels of the feature flag match that of the federated graph for which the preview is being created.
+- Make sure that the subgraphs being changed are a part of the federated graph.
+- Make sure that the changed subgraphs are a part of the config, if not present the featur subgraph for that subgraph will not be created.
+- All the subgraphs in the config have to be deployed and present on Cosmo.
+- The `cosmo.yaml` file should be present in the `.github` directory before the PR is labeled with `cosmo-preview`.
+- All the subgraphs changed in this PR should be deployed at the proposed routing_url in the CI for the preview to work.
+
+### Implementation
+
+The implementation is the same as Proposal-1, the only difference is that the feature flag is created when the PR is labeled with `cosmo-preview` and destroyed when the PR is unlabeled with `cosmo-preview` or when the pr is closed.
```

Comment:
Can you add this to the .github/cosmo.yaml as an option to activate it?
Alternatively, is there another way to configure this per repo?

Reply:
- Can you add this to the .github/cosmo.yaml as an option to activate it?
Even this can be done, by adding an enable option in the comso.yaml, but feel adding the label is much more convenient

Reply:
You're not following my idea I think.
My idea was to enable or disable the "label-mode" in the `.github/cosmo.yaml` config.
Does this make sense? Is it technically possible?
Can the user decide to opt into this in another way?

Reply:
Yep we can do that